source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -409,7 +409,7 @@ Filter::StreamOpenState Filter::openStream() {
     return StreamOpenState::IgnoreError;
   }
 
-  if (!config().grpcService().has_value()) {
+  if (!grpc_service_.has_envoy_grpc() && !grpc_service_.has_google_grpc()) {
```

Comment:
This has slight meaning change recommend leaving it out of this PR.  Same as other places.

Reply:
Still confused. Could you please share more details on "test fails as they incorrectly try to cast the HTTP service to gRPC"?

Reply:
@yanjunxiang-google Form my understanding the issue is with the per-route overrides where the client type is determined at the filter creation time based on the main configuration and runtime logic uses per-route overridden configuration. This creates a type mismatch for HTTP client with gRPC logic. I can't figure out a way to solve this with `static_cast`.

Reply:
Currently ext_proc per-route override does not support overriding gRPC service: https://github.com/envoyproxy/envoy/blob/c5f37fde4a7a5910e4426c58b614b1d79fa137b2/api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto#L450.

Do you mean config().grpcService() is causing test failures?  Could you please share which test, and what's the error message?